tests/test_library.py

```diff
@@ -113,3 +113,44 @@ def test_lookup_invalid_track(self):
     def test_search(self):
         refs = self.backend.library.search({"artist": ["abba"]})
         assert refs is not None
+
+    def test_get_images_unknown_uri(self):
+        refs = self.backend.library.get_images(["gmusic:track:unknown_uri"])
+        assert refs == {"gmusic:track:unknown_uri": []}
+
+    def test_get_images_known_uris(self):
+        self.backend.library.images = {
+            "gmusic:track:89bbf2f8-5f98-3108-a29d-b1c5e2c671a2": [
+                Image(uri="https://cdn.com/image1.jpg")
+            ],
+            "gmusic:track:fd6f9d17-63d5-3d45-b2d2-efbb5cdd4be5": [
+                Image(uri="https://cdn.com/image2.jpg")
+            ],
+            "gmusic:track:89a7bd77-3d99-3ad4-a6f6-7b9d025f02a8": [
+                Image(uri="https://cdn.com/image3.jpg")
+            ],
+            "gmusic:track:6a89747a-c2f9-3815-a8e4-6c95ebe6c083": [
+                Image(uri="https://cdn.com/image4.jpg")
+            ],
+            "gmusic:track:21535e0f-2583-3e66-9766-b0f6e180c6b0": [
+                Image(uri="https://cdn.com/image5.jpg")
+            ],
+        }
```

Comment:
Is this the right way to mock library data? I couldn't see an example of this anywhere else.
It's not great because `get_images()` doesn't really do a lot of work, so all this test is really testing is that the stored dictionary of images gets filtered to just the requested images.
If the data is mocked here in the wrong format, `get_images()` just returns that data in the wrong format. So the assertion of the how the data is formatted here doesn't necessarily even match with what Mopidy actually expects to be returned. (Right now, is does. I think)
I think, ideally, we would have a fake implementation of the gmusic API that returns a hardcoded library of data that we can work with.

Reply:
This is good a *unit* test of the `get_images()` function and its filtering. The format of the data is not important, so `[Image(uri="https://cdn.com/image1.jpg")]` might as well be replaced with `mock.sentinel.track_1_images` and so on to make it clearer that you don't care about the data structure in this specific unit test.